index.js

```diff
@@ -39,355 +32,291 @@ module.exports = Store;
- * store.hasOwn('b'); //=> true
- * store.hasOwn('c'); //=> true
- * store.hasOwn('d'); //=> true
- * store.hasOwn('foo'); //=> false
- * ```
- *
- * @param  {String} `key`
- * @return {Boolean} Returns true if `key` exists
- * @api public
- */
+  /**
+   * Assign `value` to `key` and save to disk. Can be
+   * a key-value pair or an object.
+   *
+   * ```js
+   * // key, value
+   * store.set('a', 'b');
+   * //=> {a: 'b'}
+   *
+   * // extend the store with an object
+   * store.set({a: 'b'});
+   * //=> {a: 'b'}
+   * ```
+   * @name .set
+   * @param {String} `key`
+   * @param {any} `val` The value to save to `key`. Must be a valid JSON type: String, Number, Array or Object.
+   * @return {Object} `Store` for chaining
+   * @api public
+   */
+
+  set(key, val) {
+    if (typeof key !== 'string') {
+      for (const k of Array.isArray(key) ? key : Object.keys(key)) {
+        this.set(k, key[k]);
+      }
+      return this;
+    }
+    assert.equal(typeof key, 'string', 'expected key to be a string');
+    set(this.data, key, val);
+    this.emit('set', key, val);
+    this.save();
+    return this;
+  }
 
-Store.prototype.hasOwn = function(key) {
-  var val;
-  if (key.indexOf('.') === -1) {
-    val = this.cache.hasOwnProperty(key);
-  } else {
-    val = utils.hasOwn(this.cache, key);
+  /**
+   * Get the stored `value` of `key`, or return the entire store
+   * if no `key` is defined.
+   *
+   * ```js
+   * store.set('a', {b: 'c'});
+   * store.get('a');
+   * //=> {b: 'c'}
+   *
+   * store.get();
+   * //=> {b: 'c'}
+   * ```
+   * @name .union
+   * @param  {String} `key`
+   * @return {any} The value to store for `key`.
+   * @api public
+   */
+
+  union(key, ...rest) {
+    assert.equal(typeof key, 'string', 'expected key to be a string');
+    let arr = this.get(key);
+    if (arr == null) arr = [];
+    if (!Array.isArray(arr)) arr = [arr];
+    this.set(key, unique(flatten(...arr, ...rest)));
+    return this;
   }
-  return val;
-};
 
-/**
- * Persist the store to disk.
- *
- * ```js
- * store.save();
- * ```
- * @param {String} `dest` Optionally define an alternate destination file path.
- * @api public
- */
+  /**
+   * Get the stored `value` of `key`, or return the entire store
+   * if no `key` is defined.
+   *
+   * ```js
+   * store.set('a', {b: 'c'});
+   * store.get('a');
+   * //=> {b: 'c'}
+   *
+   * store.get();
+   * //=> {b: 'c'}
+   * ```
+   *
+   * @name .get
+   * @param  {String} `key`
+   * @return {any} The value to store for `key`.
+   * @api public
+   */
+
+  get(key) {
+    return key ? get(this.data, key) : this.data;
+  }
 
-Store.prototype.save = function(dest) {
-  this.data = this.cache;
-  writeJson(dest || this.path, this.cache, this.options.indent);
-  return this;
-};
+  /**
+   * Returns `true` if the specified `key` has a value.
+   *
+   * ```js
+   * store.set('a', 'b');
+   * store.set('c', null);
+   * store.has('a'); //=> true
+   * store.has('c'); //=> false
```

Comment:
Based on the code, this should return `true` since `typeof null !== 'undefined'`